File: multiple.js
// Find the sum of all the multiples of 3 or 5 below 1000.

function isMultiple(i,numbers){
   
  return   numbers.some(n => i % n == 0);
}

function * multiple(start,end,...numbers){
    let i = start;
    console.log(numbers);
    while ( i <= end){
        if (isMultiple(i,numbers))
            {console.log(i);
            yield i;}
            i++;
    }
}

function sumMultiple(start,end){
 return  [...multiple(start,end,3,5)].reduce( (c,x) => c + x);
}

File: multiple.rb
# input limit
#return number multiple 3 or 5 between 0 and limit

def sum_multiple(limit, * numbers  )
  (2..limit).to_a.find_all{ |x| numbers.any? {|n| x % n == 0 or x % n == 0} }.reduce(:+)
end


def multiple_three_five(limit)
  sum_multiple(limit,3,5)
end
